Src/ReadBatteryVoltage.c

```diff
@@ -0,0 +1,37 @@
+#include "stm32f4xx.h"
+#include "stm32f4xx_hal_conf.h"
+#include "cmsis_os.h"
+
+#include "ReadBatteryVoltage.h"
+
+#include "Data.h"
+
+static int READ_BATTERY_VOLTAGE_PERIOD = 250;
+
+static const int BATTERY_VOLTAGE_ADC_POLL_TIMEOUT = 50;
+
+void readBatteryVoltageTask(void const* arg) {
+    BatteryVoltageData* data = (BatteryVoltageData*) arg;
+    uint32_t prevWakeTime = osKernelSysTick();
+
+    double batteryVoltageValue = 0; //Not quite sure if it's a double or int can confirm
```

Comment:
`uint32_t HAL_ADC_GetValue(ADC_HandleTypeDef* hadc)`
The `HAL_ADC_GetValue` function returns a uin32_t. We're gonna need to research/test to figure out how to scale it to mV. (i.e. is the returned value in volts? or 0x00000000 to 0xFFFFFFFF based off of 0V-5V? etc)